### PR TITLE
Issue with stdlib_sorting

### PR DESCRIPTION
**Pull request by jvdp1**
The option `-fcheck=all` of gfortran triggered a problem with the suboutine `insertion_sort`, when `i` becomes equal to `0`, and tried to access `array( i-1 )` (while the lower bound of `array` is `0`.

The solution copied what was already implemented in `stdlib_sorting_sort.fypp`.

### CODE REVIEW

**Comment by jvdp1** [comment]
To review: @wclodius2

File: src/tests/sorting/test_sorting.f90 [hunk 1]
```diff
@@ -14,7 +14,7 @@ program test_sorting
     integer(int32), parameter :: char_size = 26**4
     integer(int32), parameter :: string_size = 26**3
     integer(int32), parameter :: block_size = test_size/6
-    integer, parameter        :: repeat = 8
+    integer, parameter        :: repeat = 1
```
Comment (gareth-nx) [inline src/tests/sorting/test_sorting.f90:17]: Is this change deliberate?

**Review by gareth-nx — COMMENTED** [review]
Looks good to me, except (possibly) for the change in `repeat`, which I don't understand.

File: src/tests/sorting/test_sorting.f90 (hunk 1 above)
Comment (wclodius2) [reply src/tests/sorting/test_sorting.f90:17]: FWIW this decreases the number of times the sorts are repeated, making the test complete faster, but  potentially resulting in more noise in the comparison of the different tests.

**Review by wclodius2 — APPROVED** [review]
The changes to `stdlib_sorting_ord_sort.fypp`and `stdlib_sorting_sort_index.fypp` are all appropriate. The change to `repeat` in `test_sorting.f90` is  useful if the runtime is too long. The rest of the changes to test the changes the other codes are appropriate.

File: src/tests/sorting/test_sorting.f90 (hunk 1 above)
Comment (jvdp1) [reply src/tests/sorting/test_sorting.f90:17]: It was to speed up a it the tests. I forgot to revert it. Thank you for mentioning it. I wil revert it.

Comment (jvdp1) [inline src/tests/sorting/test_sorting.f90:17]:
```suggestion
    integer, parameter        :: repeat = 8
```

**Review by awvwgk — APPROVED** [review]
Looks good to me

**Comment by jvdp1** [comment]
Could someone with  write access review this PR that fixes a bug in stdlib_sort?

**Review by LKedward — APPROVED** [review]
Cheers, LGTM 👍

**Comment by milancurcic** [comment]
Reviews are positive, so I'll merge. Thanks!
